File: java/src/main/java/chapter12/stubbing/TestedObject.java
package chapter12.stubbing;

/**
 * Generic object that has a collaborator.
 */
public class TestedObject {

    private Collaborator collaborator;

    public TestedObject(Collaborator collaborator) {
        this.collaborator = collaborator;
    }

    public int computeSomething() {
        return 42 * collaborator.computeAndReturnValue();
    }
}

File: csharp/DeveloperTestingSamples/Samples/Chapter12/Stubbing/TestedObject.cs
namespace Samples.Chapter12.Stubbing
{
    
    // Generic object that has a collaborator.
    public class TestedObject
    {
        private ICollaborator collaborator;

        public TestedObject(ICollaborator collaborator)
        {
            this.collaborator = collaborator;
        }

        public int ComputeSomething()
        {
            return 42 * collaborator.ComputeAndReturnValue();
        }
    }
}
